File: experiments/llm-reasoning-cost/generated_code/cpp/06_struct_methods_mini.cpp
#include <iostream>
#include <vector>

struct Point {
    int32_t x;
    int32_t y;

    int32_t squared_mag() const {
        return x * x + y * y;
    }
};

int32_t filter_and_sum(const std::vector<Point>& points, int32_t max_mag) {
    int32_t total = 0;
    for (const auto& point : points) {
        if (point.squared_mag() <= max_mag) {
            total += point.squared_mag();
        }
    }
    return total;
}

int main() {
    std::vector<Point> points = {{1, 2}, {3, 4}, {0, 1}};
    int32_t max_mag = 10;
    int32_t result = filter_and_sum(points, max_mag);
    std::cout << result << std::endl; // Output: 6
    return 0;
}

File: experiments/llm-reasoning-cost/generated_code/rust/06_struct_methods_mini.rs
#[derive(Debug)]
struct Point {
    x: i32,
    y: i32,
}

impl Point {
    fn squared_mag(&self) -> i32 {
        self.x * self.x + self.y * self.y
    }
}

fn filter_and_sum(points: Vec<Point>, max_mag: i32) -> i32 {
    let mut total = 0;
    for point in points {
        let mag = point.squared_mag();
        if mag <= max_mag {
            total += mag;
        }
    }
    total
}

fn main() {
    let points = vec![
        Point { x: 1, y: 2 },
        Point { x: 3, y: 4 },
        Point { x: 0, y: 1 },
    ];
    let max_mag = 10;
    let result = filter_and_sum(points, max_mag);
    println!("Total: {}", result); // Output: Total: 6
}
